Sig Instrumentation Spotlight (#370)
kubernetes/contributor-site: content/en/blog/2023/sig-instrumentation-spotlight.md

---
layout: blog
title: "Spotlight on SIG Instrumentation"
slug: sig-instrumentation-spotlight-2023
date: 2023-02-03

> author: "Imran Noor Mohamed(Delivery Hero)"

Reviewer: ```suggestion
author: "Imran Noor Mohamed (Delivery Hero)"
```

---

> 

Reviewer 2: Add **Author**: \<you\> in the main body.

Reviewer: ```
author: "Imran Noor Mohamed(Delivery Hero)"
```
in the front matter is fine - IMO

**Han (HK)**: We have many different sub-projects and we are in dire need of people who can come and help shepherd them. Our most important projects in-tree (that is, within the kubernetes/kubernetes repo) are metrics, tracing, and, structured logging. Our most important projects out-of-tree are (a) KSM (kube-state-metrics) and (b) metrics-server.

**Elana (EH)**: Echoing this, we would love to bring on more maintainers for kube-state-metrics and metrics-server. Our friends at WG Structured Logging are also looking for contributors. Other subprojects include klog, prometheus-adapter, and a new subproject that we just launched for collecting high-fidelity, scalable utilization metrics called usage-metrics-collector. All are seeking new contributors!

## Current status and ongoing challenges

**Imran (INM)**: For release [1.26](https://github.com/kubernetes/sig-release/tree/master/releases/release-1.26) we can see that there are a relevant number of metrics, logs, and tracing [KEPs](https://www.k8s.dev/resources/keps/) in the pipeline. Would you like to point out important things for last release (maybe alpha & stable milestone candidates?)

> **Han (HK)**: We can now generate documentation for every single metric in the main Kubernetes code base! We have a pretty fancy static analysis pipeline that enables this functionality. We’ve also added feature metrics so that you can look at your metrics to determine which features are enabled in your cluster at a given time. Lastly, we added a component-sli endpoint, which should make it easy for people to create availability SLOs for *control-plane* components.

Reviewer 2: Link to https://kubernetes.io/docs/reference/instrumentation/metrics/ ?

Author: awesome! Good to plug them

**Elana (EH)**: I think SIG Instrumentation is a really interesting SIG because it poses different kinds of opportunities to get involved than in other SIGs. You don’t have to be a software developer to contribute to our SIG! All of our components and subprojects are focused on better understanding Kubernetes and its performance in production, which allowed me to get involved as one of the few SIG Chairs working as an SRE at that time. I like that we provide opportunities for newcomers to contribute through using, testing, and providing feedback on our subprojects, which is a lower barrier to entry. Because many of these projects are out-of-tree, I think one of our challenges is to figure out what’s in scope for core Kubernetes SIGs instrumentation subprojects, what’s missing, and then fill in the gaps.

## Community and contribution

> **Imran (INM)**: Kubernetes values community over products.  Any recommendation for anyone looking into getting involved in SIG Instrumentation work? Where should they start (new contributor-friendly areas within SIG?)

Reviewer 2: Link to https://github.com/kubernetes/community/tree/master/sig-instrumentation#meetings ?

---
layout: blog
title: "Spotlight on SIG Instrumentation"

> slug: sig-instrumentation-spotlight

Reviewer: ```suggestion
slug: sig-instrumentation-spotlight-2023
```

date: 2023-01-31

> canonicalUrl: https://www.kubernetes.dev/blog/2023/01/31/sig-instrumentation-spotlight-2023/

Reviewer: Don't set a `canonicalUrl` for **this** PR. However, the article on https://blog.k8s.io/ does need that front matter variable set.

---

> **Author**: Imran Noor Mohamed(Delivery Hero)

Reviewer: For this site, AIUI we prefer to use the front matter to set an author. See other articles such as https://github.com/kubernetes/contributor-site/blame/master/content/en/blog/2022/sig-storage-spotlight.md#L6 for an example.

> Observability requires the right data at the right time for the right consumer (human or piece of software) to make the right decision. In the context of Kubernetes, having best practices for cluster observability across all kubernetes components is crucial. 

Reviewer: ```suggestion
Observability requires the right data at the right time for the right consumer (human or piece of software) to make the right decision. In the context of Kubernetes, having best practices for cluster observability across all components is crucial.
```

> SIG Instrumentation helps to address this issue by providing best practices and tools that all other SIG uses to instrument kubernetes components like the *Kube API server*, *scheduler*, and *controller manager*.

Reviewer: ```suggestion
SIG Instrumentation helps to address this issue by providing best practices and tools that all other SIGs use to instrument Kubernetes components–like the *API server*, *scheduler*, *kubelet* and *controller manager*.
```
?

Author: done

> In this SIG Instrumentation spotlight, [Imran Noor Mohamed](https://www.linkedin.com/in/imrannoormohamed/), SIG Contribex-Comms tech lead talked with [Elana Hashman](https://twitter.com/ehashdn), and [Han Kang](https://www.linkedin.com/in/hankang), Chairs of SIG instrumentation, on how the SIG is organized, what are the current challenges and how anyone can get involved and contribute. 

Reviewer: ```suggestion
In this SIG Instrumentation spotlight, [Imran Noor Mohamed](https://www.linkedin.com/in/imrannoormohamed/), SIG ContribEx-Comms tech lead talked with [Elana Hashman](https://twitter.com/ehashdn), and [Han Kang](https://www.linkedin.com/in/hankang), chairs of SIG Instrumentation, on how the SIG is organized, what are the current challenges and how anyone can get involved and contribute. 
``

Author: done

## About SIG Instrumentation

**Imran (INM)**: Hello, thank you for the opportunity of learning more about SIG Instrumentation. Could you tell us a bit about yourself, your role, and how you got involved in SIG Instrumentation?

**Han (HK)**: I started in SIG Instrumentation in 2018, and became a chair in 2020. I primarily got involved with SIG instrumentation due to a number of upstream issues with metrics which ended up affecting GKE in bad ways. As a result, we ended up launching an initiative to stabilize our metrics and make metrics a proper API.

**Elana (EH)**: I also joined SIG Instrumentation in 2018 and became a chair at the same time as Han. I was working as a site reliability engineer (SRE) on bare metal Kubernetes clusters and was working to build out our observability stack. I encountered some issues with label joins where Kubernetes metrics didn’t match Kube-state-metrics (KSM) and started participating in SIG meetings to improve things. I helped test performance improvements to Kube-state-metrics and ultimately coauthored a KEP for overhauling metrics in the 1.14 release to improve usability.

> **Imran (INM)**: Interesting! Does that mean SIG-Instrumentation involves a lot of plumbing?

Reviewer: ```suggestion
**Imran (INM)**: Interesting! Does that mean SIG Instrumentation involves a lot of plumbing?
```

Author: done

**Han (HK)**: I wouldn’t say it involves a ton of plumbing, though it does touch basically every code base. We have our own dedicated directories for our metrics, logs, and tracing frameworks which we tend to work out of primarily. We do have to interact with other SIGs in order to propagate our changes which makes us more of a horizontal SIG.

> **Imran (INM)**: Speaking about interaction and coordination with other SIG could you describe how the SIG is organized?

Reviewer: Was this actually:
```suggestion
**Imran (INM)**: Speaking about interaction and coordination with other SIGs could you describe how the SIG is organized?
```

Author: yes, that was the intention. I got confused between plural and singular forms.
